### DISCUSSION TITLE
Long File Name(s)

### DISCUSSION THREAD

**Discussion by MRK6310** [WinMerge/winmerge#2323] [General]
When trying to copy files from one drive to another I get an error report: 'The file name(s) would be too long for the destination folder'. Is there a way to overcome this without renaming all the files concerned? I'm using Windows 10 on a 64-bit PC, Winmerge 2.14.0.

**Comment by sdottaka** [comment]
WinMerge relies on SHFileOperation() and IFileOperation to copy files, and unfortunately, these do not seem to support long file names.

If the folder name is long, you may be able to use the subst command to assign long folder names to drives, and then compare and copy between those drives.

~~~
subst C:\long folder name\ Y:
subst D:\long folder name\ Z:
~~~

refs #2100
